File: mns-core/src/test/java/cz/fb/manaus/core/category/CategoryServiceTest.java
package cz.fb.manaus.core.category;

import cz.fb.manaus.core.MarketCategories;
import cz.fb.manaus.core.category.categorizer.SportCategorizer;
import cz.fb.manaus.core.model.EventType;
import cz.fb.manaus.core.model.Market;
import cz.fb.manaus.core.model.Side;
import cz.fb.manaus.core.test.AbstractLocalTestCase;
import cz.fb.manaus.core.test.CoreTestFactory;
import org.junit.Before;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Set;

import static org.hamcrest.CoreMatchers.hasItem;
import static org.hamcrest.CoreMatchers.hasItems;
import static org.junit.Assert.assertThat;

public class CategoryServiceTest extends AbstractLocalTestCase {
    public static final String SPORT_SOCCER = Category.MARKET_PREFIX + SportCategorizer.PREFIX + MarketCategories.SOCCER;
    public static final String SPORT_TENNIS = Category.MARKET_PREFIX + SportCategorizer.PREFIX + MarketCategories.TENNIS;
    @Autowired
    private CategoryService categoryService;
    private Market market;
    private EventType eventType;

    @Before
    public void createMarket() {
        market = CoreTestFactory.newMarket();
        eventType = new EventType("1", "Y");
        market.setEventType(eventType);
    }

    @Test
    public void testCategory() throws Exception {
        eventType.setName("Soccer");
        Set<String> categories = categoryService.getMarketCategories(market, false);
        assertThat(categories, hasItem(SPORT_SOCCER));

        eventType.setName("Tennis");
        categories = categoryService.getMarketCategories(market, false);
        assertThat(categories, hasItem(SPORT_TENNIS));

        eventType.setName("Horse Racing");
        categories = categoryService.getMarketCategories(market, false);
        assertThat(categories, hasItem(Category.MARKET_PREFIX + SportCategorizer.PREFIX + MarketCategories.HORSES));

        eventType.setName("Golf");
        categories = categoryService.getMarketCategories(market, false);
        assertThat(categories, hasItem(Category.MARKET_PREFIX + SportCategorizer.PREFIX + MarketCategories.GOLF));
    }


    @Test
    public void testBetCategory() throws Exception {
        Set<String> categories = categoryService.getSettledBetCategories(
                CoreTestFactory.newSettledBet(2d, Side.LAY), false, BetCoverage.EMPTY);
        assertThat(categories,
                hasItems("market_country_br", "market_runnerCount_3", "market_sport_soccer", "market_type_match_odds"));
    }
}

File: mns-core/src/test/java/cz/fb/manaus/core/category/CategoryServiceTest.kt
package cz.fb.manaus.core.category

import cz.fb.manaus.core.MarketCategories
import cz.fb.manaus.core.category.categorizer.SPORT_PREFIX
import cz.fb.manaus.core.model.market
import cz.fb.manaus.core.model.realizedBet
import cz.fb.manaus.core.test.AbstractLocalTestCase
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import kotlin.test.assertTrue

class CategoryServiceTest : AbstractLocalTestCase() {
    @Autowired
    private lateinit var categoryService: CategoryService

    @Test
    fun `market category`() {
        val eventType = market.eventType
        var market = market

        market = market.copy(eventType = eventType.copy(name = "Soccer"))
        assertTrue { SPORT_SOCCER in categoryService.getMarketCategories(market, false) }

        market = market.copy(eventType = eventType.copy(name = "Tennis"))
        assertTrue { SPORT_TENNIS in categoryService.getMarketCategories(market, false) }

        market = market.copy(eventType = eventType.copy(name = "Golf"))
        assertTrue { Category.MARKET_PREFIX + SPORT_PREFIX + MarketCategories.GOLF in categoryService.getMarketCategories(market, false) }
    }

    @Test
    fun `bet category`() {
        val categories = categoryService.getRealizedBetCategories(realizedBet, false)
        assertTrue { categories.containsAll(listOf("market_country_cz", "market_sport_soccer", "market_type_match_odds")) }
    }

    companion object {
        const val SPORT_SOCCER = Category.MARKET_PREFIX + SPORT_PREFIX + MarketCategories.SOCCER
        const val SPORT_TENNIS = Category.MARKET_PREFIX + SPORT_PREFIX + MarketCategories.TENNIS
    }
}
